packages/runtime/src/templates/edge-shared/rsc-data.ts

```diff
@@ -60,6 +60,11 @@ export const getRscDataRouter = ({ routes: staticRoutes, dynamicRoutes }: Preren
     const debug = request.headers.has('x-next-debug-logging')
     const log = debug ? (...args: unknown[]) => console.log(...args) : noop
     const url = new URL(request.url)
+
+    // Set the 'vary' header to 'RSC' to ensure that we cache correctly for the different
+    // possible content-types: application/octet-stream and text/html
+    request.headers.set('vary', 'RSC')
```

Comment:
Shouldn't this be on the Response?

Reply:
Yeah, you're right. Not sure why I put it on the request (late night coding 🙃 ), and also not sure why my tests and `curl`s pass. 🤔

Reply:
It already worked. The problem isn't it returning the wrong data - that was already fixed. The problem is the browser caching the wrong data. The tests should be checking that the server is returning the `Vary` header, and you can manually test it in the browser with the instructions in https://github.com/netlify/pod-ecosystem-frameworks/issues/352